tests/ui/checked_unwrap/simple_conditionals.rs

```diff
@@ -78,3 +78,24 @@ fn main() {
 
     assert!(x.is_ok(), "{:?}", x.unwrap_err()); // ok, it's a common test pattern
 }
+
+mod issue_5579 {
```

Comment:
Can you move this test in its own file in the crashes dir: `tests/ui/crashes/ice-5579.rs`